felix/bpf/bpf.go

```diff
@@ -1229,13 +1229,15 @@ func (b *BPFLib) GetXDPMode(ifName string) (XDPMode, error) {
 
 	s := strings.Fields(string(output))
 	allModes := map[string]XDPMode{
-		XDPDriver.String():  XDPDriver,
-		XDPOffload.String(): XDPOffload,
-		XDPGeneric.String(): XDPGeneric,
-	}
-	for i := range s {
-		if mode, ok := allModes[s[i]]; ok {
-			return mode, nil
+		"xdp":        XDPDriver, // We write "xdpdrv" but read back "xdp"
+		"xdpoffload": XDPOffload,
+		"xdpgeneric": XDPGeneric,
+	}
+	for _, mode := range []string{"xdpgeneric", "xdpoffload", "xdp"} {
```

Comment:
How about `for modeString, mode := range allModes { ... }` and `if f == modeString { return mode, nil }` ?  That would avoid repeating the strings that we're looking for?

Reply:
That was to ensure deterministic ordering in case both strings showed up in the output.  Will rework to make that clear.